### PR TITLE
Fixes #29504 - Skip plugin registration if the DB is not ready

### PR DESCRIPTION
**Pull request by kamils-iRonin**

### CODE REVIEW

**Review by timogoebel — DISMISSED** [review]

File: app/registries/foreman/plugin.rb [hunk 1]
```diff
@@ -79,6 +79,8 @@ def register(id, &block)
         plugin.after_initialize
 
         registered_plugins[id] = plugin
+      rescue ActiveRecord::NoDatabaseError, ActiveRecord::StatementInvalid => e
```
Comment (timogoebel) [inline app/registries/foreman/plugin.rb]: Looks like rubocop complaints:

```
app/registries/foreman/plugin.rb:82:7: W: Lint/ShadowedException: Do not shadow rescued Exceptions.
      rescue ActiveRecord::NoDatabaseError, ActiveRecord::StatementInvalid => e ...
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment (kamils-iRonin) [reply app/registries/foreman/plugin.rb]: actually, `ActiveRecord::NoDatabaseError` is redundant as it inherits from `ActiveRecord::StatementInvalid`

https://github.com/rails/rails/blob/09ff877610a09d930ca84a94113f678d4e79e100/activerecord/lib/active_record/errors.rb#L186-L188

**Review by timogoebel — APPROVED** [review]

**Comment by timogoebel** [comment]
Katello tests fail and this might be related. :-(

**Comment by mmoll** [comment]
[test katello]

**Comment by mmoll** [comment]
all :green_heart:

**Comment by timogoebel** [comment]
Oh, nice!

**Comment by timogoebel** [comment]
Thanks, @kamils-iRonin.

**Pull request merged**